wallet-test/src/test/scala/org/bitcoins/wallet/WalletIntegrationTest.scala

```diff
@@ -110,28 +111,28 @@ class WalletIntegrationTest extends BitcoinSWalletTest {
       _ = assert(outgoingTx.isDefined)
       _ = assert(outgoingTx.get.inputAmount == valueFromBitcoind)
       _ = assert(outgoingTx.get.sentAmount == valueToBitcoind)
-      _ = assert(outgoingTx.get.feeRate == feeRate)
-      _ = assert(outgoingTx.get.expectedFee == feeRate.calc(signedTx))
       _ = assert(
-        WalletTestUtil.isCloseEnough(feeRate.calc(signedTx),
-                                     outgoingTx.get.actualFee,
-                                     3.satoshi))
+        WalletTestUtil.isFeeRateCloseEnough(outgoingTx.get),
+        s"Actual fee=${outgoingTx.get.feeRate} expectedFeeRate=$feeRate"
+      )
+      balancePostSend <- wallet.getBalance()
+    } yield {
+      assert(outgoingTx.get.expectedFee == feeRate.calc(signedTx))
+      assert(
+        outgoingTx.get.expectedFee === outgoingTx.get.actualFee +- feeRate.currencyUnit, // could be off by one due to rounding
```

Comment:
off by one unit in fee measurement? I.e. `+- 40 sat/vbyte` if `40 sat/vbyte` was your original fee estimate?

Reply:
No, in the measurement of the actual fee paid, `expectedFee` and `actualFee` are `CurrencyUnit` values. So `+- 40 sats` in the fee paid if the we used a fee rate of `40 sat/byte`